dpx/src/specials/util.rs

```diff
@@ -764,11 +766,11 @@ pub(crate) unsafe fn spc_util_read_blahblah(
                     }
                 }
                 "page" => {
-                    let mut page: f64 = 0.;
-                    if !page_no.is_null() && spc_util_read_numbers(&mut page, 1i32, ap) == 1i32 {
-                        *page_no = page as i32
+                    let page: ArrayVec<[_; 1]> = read_numbers(ap);
+                    if !page_no.is_null() && page.len() == 1 {
```

Comment:
Wrong sequence of calculations.
`read_numbers` has not be calculated if `page_no.is_null()`